tests/scripts/test_trace_docs_deps.py

```diff
@@ -0,0 +1,93 @@
+from typing import List
+
+import py.path
+
+from scripts import find_docusaurus_refs, get_import_paths, get_local_imports
+
+
+def test_find_docusaurus_refs_parses_correctly(tmpdir_factory):
+    docs_dir: py.path.local = tmpdir_factory.mktemp("docs")
+    temp_file: py.path.local = docs_dir.join("file.md")
+    with open(temp_file, "w") as f:
+        f.write(
+            "```python file=../../../../../../../../../../../../../../../a/b/c/script1.py#L1\n"
+        )
+        f.write(
+            "```python file=../../../../../../../../../../../../../../../d/e/f/script2.py#L5-10\n"
+        )
+        f.write(
+            "```yaml file=../../../../../../../../../../../../../../../g/h/i/script3.py#L100-200"
+        )
+
+    assert len(find_docusaurus_refs(str(docs_dir))) == 3
+
+
+def test_find_docusaurus_refs_does_not_duplicate(tmpdir_factory):
+    docs_dir: py.path.local = tmpdir_factory.mktemp("docs")
+    temp_file: py.path.local = docs_dir.join("temp_file.md")
+    with open(temp_file, "w") as f:
+        f.write(
+            "```python file=../../../../../../../../../../../../../../../a/b/c/script_a.py#L1\n"
+        )
+        f.write(
+            "```python file=../../../../../../../../../../../../../../../a/b/c/script_a.py#L5-10"
+        )
```

Comment:
@cdkini Can there be a way to use the file_relative_path() type utility we have in order to avoid keeping track of the directory levels?